packages/maskbook/src/utils/native-rpc/types.ts

```diff
@@ -1,9 +1,13 @@
+import type { PersonaWithProviders } from '../../database'
 /**
  * JSON RPC calls that can be called from the native side.
  * JS = server, native = client
  */
 export interface WebviewAPIs {
     web_echo<T>(arg: T): Promise<T>
+    getUrl(path: string): string
+    getStorage(key: string): Promise<browser.storage.StorageObject>
```

Comment:
This should not be a public API. What exact information do you want to expose? Please provide an API for that purpose instead of public a method for querying our internal state.

Reply:
App native plans to retrieve several setting values ("launchPage", etc ..) from JS side.
To ensure security JS could define a whitelist of `storage keys`, but that does not need to be present in the name of the API IMO

Reply:
So the implementation should not match the browser.storage.local.get. It should get value from our settings object.

Reply:
done